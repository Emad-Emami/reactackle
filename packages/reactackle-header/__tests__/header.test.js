'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  Header,
  HeaderRegion,
  HeaderTitle,
} from '../src';

jest.mock('react-dom');
jest.mock('react-resize-detector', () => jest.fn(() => <div />));

const context = {
  createNodeMock(element) {
    if (element.type !== 'header') return null;
    
    return {
      getBoundingClientRect() {
        return {
          bottom: 124,
          height: 124,
          left: 0,
          right: 485,
          top: 0,
          width: 485,
        };
      },
    };
  },
};

const map = {};

window.removeEventListener = jest.fn();
window.addEventListener = jest.fn((event, cb) => {
  map[event] = cb;
});

describe('<Header/>', () => {
  it('renders correctly with default props', () => {
    const component = mount(
      <Header />,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('call removeEventListener if unmount and header is condenses', () => {
    const component = mount(
      <Header condenses />,
    );
    
    component.instance().componentWillUnmount();
    
    expect(window.removeEventListener).toHaveBeenCalledWith(
      'scroll',
      map.scroll,
    );
  });
  
  it('renders correctly if change prop behavior set fixed', () => {
    const component = mount(
      <Header behavior="static" />,
    );
    
    expect(toJson(component)).toMatchSnapshot();
    component.instance().componentWillReceiveProps({ behavior: 'fixed' });
    expect(window.removeEventListener).toHaveBeenCalledWith(
      'scroll',
      map.scroll,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if change prop behavior set static', () => {
    const component = mount(
      <Header behavior="fixed" />,
    );
    
    expect(toJson(component)).toMatchSnapshot();
    component.instance().componentWillReceiveProps({ behavior: 'static' });
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if change prop condenses', () => {
    const component = mount(
      <Header condenses />,
    );
    
    expect(toJson(component)).toMatchSnapshot();
    component.instance().componentWillReceiveProps({ condenses: false });
    expect(window.removeEventListener).toHaveBeenCalledWith(
      'scroll',
      map.scroll,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if prop behavior set static', () => {
    const component = mount(
      <Header behavior="static" />,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if prop behavior set fixed', () => {
    const component = mount(
      <Header behavior="fixed" />,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if prop size set default', () => {
    const component = mount(
      <Header size="default" />,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if prop size set blank', () => {
    const component = mount(
      <Header size="blank" />,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if prop size set dense', () => {
    const component = mount(
      <Header size="dense" />,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if have children label with id test', () => {
    const component = mount(
      <Header>
        <label id="test" />
      </Header>,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if backgroundColor set', () => {
    const component = mount(
      <Header backgroundColor="red" />,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if backgroundImage set', () => {
    const component = mount(
      <Header backgroundImage="example.png" />,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('renders correctly if behavior set type-2 and page scroll', () => {
    const component = mount(
      <Header
        condensedHeaderHeight={50}
        condenses
        keepCondensedHeader
        behavior="fixed"
      >
        <HeaderRegion spread>
          <HeaderTitle>
            <div style={{ minHeight: '100px' }}>
              Header behavior set type-2
            </div>
          </HeaderTitle>
        </HeaderRegion>
      </Header>,
      context,
    );
    
    expect(toJson(component)).toMatchSnapshot();
    
    window.pageYOffset = 200;
    map.scroll();
    
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe('<HeaderRegion/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <HeaderRegion />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});

describe('<HeaderTitle/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <HeaderTitle />,
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});
