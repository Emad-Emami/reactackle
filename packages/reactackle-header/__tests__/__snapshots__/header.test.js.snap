// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders correctly if backgroundColor set 1`] = `
<Header
  backgroundColor="red"
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor="red"
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP hitBWR"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor="red"
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ gzBRIK"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if backgroundImage set 1`] = `
<Header
  backgroundColor=""
  backgroundImage="example.png"
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderBackgroundContainerStyled
            style={
              Object {
                "backgroundImage": "url(example.png)",
              }
            }
          >
            <div
              className="HeaderBackgroundContainerStyled-ISVpt hKbsvE"
              style={
                Object {
                  "backgroundImage": "url(example.png)",
                }
              }
            />
          </HeaderBackgroundContainerStyled>
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if behavior set type-2 and page scroll 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="fixed"
  condensedHeaderHeight={50}
  condenses={true}
  keepCondensedHeader={true}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="fixed"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ gdZgSl"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            >
              <HeaderRegion
                horizontalAlign="start"
                size="default"
                spread={true}
                verticalAlign="stretch"
              >
                <HeaderRegionStyled
                  horizontalAlign="start"
                  size="default"
                  spread={true}
                  verticalAlign="stretch"
                >
                  <div
                    className="HeaderRegionStyled-eAGsQk kOlVKp"
                    size="default"
                  >
                    <HeaderTitle
                      colorScheme="light"
                      size="default"
                    >
                      <HeaderTitleStyled
                        colorScheme="light"
                        size="default"
                      >
                        <div
                          className="HeaderTitleStyled-epPCcX kNAejw"
                          size="default"
                        >
                          <div
                            style={
                              Object {
                                "minHeight": "100px",
                              }
                            }
                          >
                            Header behavior set type-2
                          </div>
                        </div>
                      </HeaderTitleStyled>
                    </HeaderTitle>
                  </div>
                </HeaderRegionStyled>
              </HeaderRegion>
            </div>
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if behavior set type-2 and page scroll 2`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="fixed"
  condensedHeaderHeight={50}
  condenses={true}
  keepCondensedHeader={true}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="fixed"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ gdZgSl"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            >
              <HeaderRegion
                horizontalAlign="start"
                size="default"
                spread={true}
                verticalAlign="stretch"
              >
                <HeaderRegionStyled
                  horizontalAlign="start"
                  size="default"
                  spread={true}
                  verticalAlign="stretch"
                >
                  <div
                    className="HeaderRegionStyled-eAGsQk kOlVKp"
                    size="default"
                  >
                    <HeaderTitle
                      colorScheme="light"
                      size="default"
                    >
                      <HeaderTitleStyled
                        colorScheme="light"
                        size="default"
                      >
                        <div
                          className="HeaderTitleStyled-epPCcX kNAejw"
                          size="default"
                        >
                          <div
                            style={
                              Object {
                                "minHeight": "100px",
                              }
                            }
                          >
                            Header behavior set type-2
                          </div>
                        </div>
                      </HeaderTitleStyled>
                    </HeaderTitle>
                  </div>
                </HeaderRegionStyled>
              </HeaderRegion>
            </div>
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if change prop behavior set fixed 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if change prop behavior set fixed 2`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if change prop behavior set static 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="fixed"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="fixed"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ gdZgSl"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if change prop behavior set static 2`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="fixed"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="fixed"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ gdZgSl"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if change prop condenses 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={true}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if change prop condenses 2`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={true}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if have children label with id test 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            >
              <label
                id="test"
              />
            </div>
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if prop behavior set fixed 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="fixed"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="fixed"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ gdZgSl"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if prop behavior set static 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if prop size set blank 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="blank"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="blank"
  >
    <header
      className="HeaderStyled-ePhNYP flzWCo"
      size="blank"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="blank"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ lftLLN"
          size="blank"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if prop size set default 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly if prop size set dense 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="dense"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="dense"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="dense"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="dense"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="dense"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<Header/> renders correctly with default props 1`] = `
<Header
  backgroundColor=""
  backgroundImage=""
  behavior="static"
  condensedHeaderHeight={0}
  condenses={false}
  keepCondensedHeader={false}
  onChangeOffset={[Function]}
  size="default"
>
  <HeaderStyled
    customBackgroundColor=""
    innerRef={[Function]}
    size="default"
  >
    <header
      className="HeaderStyled-ePhNYP ficYZQ"
      size="default"
    >
      <HeaderWrapperStyled
        behavior="static"
        customBackgroundColor=""
        innerRef={[Function]}
        size="default"
      >
        <div
          className="HeaderWrapperStyled-dyDPuJ fFVBOU"
          size="default"
        >
          <HeaderContentWrapperStyled>
            <div
              className="HeaderContentWrapperStyled-hsJsJh dyxJvl"
            />
          </HeaderContentWrapperStyled>
        </div>
      </HeaderWrapperStyled>
      <mockConstructor
        handleWidth={true}
        onResize={[Function]}
      >
        <div />
      </mockConstructor>
    </header>
  </HeaderStyled>
</Header>
`;

exports[`<HeaderRegion/> renders correctly with default props 1`] = `
<div
  className="HeaderRegionStyled-eAGsQk MCzkm"
  size="default"
/>
`;

exports[`<HeaderTitle/> renders correctly with default props 1`] = `
<div
  className="HeaderTitleStyled-epPCcX kNAejw"
  size="default"
/>
`;
